File: src/MaxStackElement.pas
(* Enter your code here. Read input from STDIN. Print output to STDOUT *)
program Solution;

var
    stack : array of longint;
    N, val, pointer, i, currMax : longint;
    query : byte;

begin
    readln(N);
    setlength(stack, N);
    pointer := 0;
    for i := 0 to N-1 do
    begin
        read(query);
        if query = 1 then 
        begin
            read(val);
            if pointer = 0 then
                stack[pointer] := val
            else 
            begin
                currMax := stack[pointer-1];
                if val > currMax then 
                    stack[pointer] := val
                else
                    stack[pointer] := currMax;
            end;
            Inc(pointer);
        end
        else if query = 2 then
            Dec(pointer)
        else 
            writeln(stack[pointer-1]);
    end;
end.

File: src/MaxStackElement.py
# Enter your code here. Read input from STDIN. Print output to STDOUT
N = int(raw_input())
stack = []
for i in range(N):
    line = raw_input()
    query = line[0]
    if query == '1':
        val = int(line[2:])
        if len(stack) == 0:
            stack.append(val)
        else:
            currMax = stack[-1]
            stack.append(val if val > currMax else currMax)
    elif query == '2':
        stack.pop()
    else:
        print(stack[-1])
